### PR TITLE
[REG] fix Issue 14861 - Error in stdio.d in LockingTextReader.readFront()

### PR DESCRIPTION
**Pull request by WalterBright**
This removes the decoding misfeature of LockingTextReader. Also, the call to ungetc() is moved to the destructor, so it only gets called once.

https://issues.dlang.org/show_bug.cgi?id=14861

The earlier attempt:
https://github.com/D-Programming-Language/phobos/pull/3622

Alternate fix proposal:
https://github.com/D-Programming-Language/phobos/pull/3696

### CODE REVIEW

File: std/stdio.d [hunk 1]
```diff
@@ -2984,6 +2942,24 @@ unittest // bugzilla 12320
...
+        fw.writeln(i, ";", "Иванов;Пётр;Петрович");
+    fw.close();
+    // Test read
+    File fr = File(deleteme, "r");
+    int nom; string fam, nam, ot;
+    // Error format read
+    while(!fr.eof)
+        fr.readf("%s;%s;%s;%s\n", &nom, &fam, &nam, &ot);
+    fr.close();
+    std.file.remove(deleteme);
```
Comment (DmitryOlshansky) [inline std/stdio.d]: should be in scope(exit) right after creation

**Comment by WalterBright** [comment]
This is a regression fix. It needs pulling.

**Comment by DmitryOlshansky** [comment]
> This is a regression fix. It needs pulling

Open against the stable branch then?

All in all it looks good.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**